phys2denoise.py

```diff
@@ -0,0 +1,234 @@
+                                 f'to run {metric}')
+            else:
+                args[param.name] = param.default
+        else:
+            args[param.name] = metric_args[param.name]
+
+    return args
+
+
+def print_metric_call(metric, args):
+    """
+    Log a message to describe how a metric is being called.
+
+    Parameters
+    ----------
+    metric : function
+        Metric function that is being called
+    args : dict
+        Dictionary containing all arguments that are used to parametrise metric
+
+    Notes
+    -----
+    Outcome
+        An info-level message for the logger.
+    """
+    msg = f'The {metric} regressor will be computed using the following parameters:'
+
+    for arg in args:
+        msg = f'{msg}\n    {arg} = {args[arg]}'
+
+    msg = f'{msg}\n'
+
+    LGR.info(msg)
+
+
+@due.dcite(
+     Doi(''),
+     path='phys2denoise',
+     description='Creation of regressors for physiological denoising',
+     version=__version__,
+     cite_module=True)
+def phys2denoise(filename, metric_args, outdir='.',
+                 metrics=[crf, rpv, rv, rvt, rrf, 'retroicor_card', 'retroicor_resp'],
+                 debug=False, quiet=False):
+    """
+    Run main workflow of phys2denoise.
+
+    Runs the parser, does some checks on input, then computes the required metrics.
+
+    Notes
+    -----
+    The code was greatly copied from phys2bids (copyright the physiopy community)
+
+    """
+    # Check options to make them internally coherent pt. I
+    # #!# This can probably be done while parsing?
+    outdir = os.path.abspath(outdir)
+    os.makedirs(outdir)
+    os.makedirs(os.path.join(outdir, 'code'))
+    log_path = os.path.join(outdir, 'code', 'logs')
+    os.makedirs(log_path)
+
+    # Create logfile name
+    basename = 'phys2denoise_'
+    extension = 'tsv'
+    isotime = datetime.datetime.now().strftime('%Y-%m-%dT%H%M%S')
+    logname = os.path.join(log_path, (basename + isotime + '.' + extension))
+
+    # Set logging format
+    log_formatter = logging.Formatter(
+        '%(asctime)s\t%(name)-12s\t%(levelname)-8s\t%(message)s',
+        datefmt='%Y-%m-%dT%H:%M:%S')
+
+    # Set up logging file and open it for writing
+    log_handler = logging.FileHandler(logname)
+    log_handler.setFormatter(log_formatter)
+    sh = logging.StreamHandler()
+
+    if quiet:
+        logging.basicConfig(level=logging.WARNING,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+    elif debug:
+        logging.basicConfig(level=logging.DEBUG,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+    else:
+        logging.basicConfig(level=logging.INFO,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+
+    version_number = _version.get_versions()['version']
```

Comment:
You already get the version earlier in this file, don't you?

Reply:
Possibly yes. I honestly copied it from phys2bids :sweat_smile:, that's why I'm asking who took care of this over there.
Any other solution (we'll retroactively add it to phys2bids)?

Reply:
Either you use the `__version__` you imported, or you use this line.